rpc/client/rpc_test.go

```diff
@@ -414,6 +424,18 @@ func TestTx(t *testing.T) {
 	}
 }
 
+func TestTxSearchWithTimeout(t *testing.T) {
+	// Get a client with a time-out of 10 secs.
+	timeoutClient := getHTTPClientWithTimeout(10)
```

Comment:
not sure what are we testing here. we certainly don't check that when timeout expires, TxSearch func exits.

Reply:
I found it pretty hard to actually test for the timeout, because if it's too early its the rpc request that fails, and if it's too late you just get the query. So just testing that it basically works with a timeout client, but not testing the timeout...